packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -895,9 +895,13 @@ class _RenderDecoration extends RenderBox {
     final double right = overallWidth - contentPadding.right;
 
     height = layout.containerHeight;
-    baseline = decoration.border == null || decoration.border.isOutline
-      ? layout.outlineBaseline
-      : layout.inputBaseline;
+    if (decoration.border == null && decoration.contentPadding == EdgeInsets.zero) {
```

Comment:
why is a 0 pixel border different than a 0.00001px border?

Reply:
I'm trying to detect the collapsed border case here. I should just pass along the boolean.